inst/include/dust/dust.hpp

```diff
@@ -235,6 +229,22 @@ class Dust {
   rng_t& pick_generator(const size_t i) {
     return _rng(i % _rng.size());
   }
+
+  void initialise(const init_t data, const size_t step,
+                  const size_t n_particles) {
+    _particles.clear();
+    _particles.reserve(n_particles);
+    for (size_t i = 0; i < n_particles; ++i) {
+      _particles.push_back(Particle<T>(data, step));
+    }
```

Comment:
Just overwrite
```suggestion
    for (size_t i = 0; i < n_particles; ++i) {
      _particles[i] = Particle<T>(data, step);
    }
```

Reply:
Although this won't work if  `n_particles != _particles.size()` - is that the case sometimes?

Reply:
Once created particle size is immutable